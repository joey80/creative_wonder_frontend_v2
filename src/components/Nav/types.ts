interface NavLinks {
  name: string;
  children?: Array<NavLinks>;
}

interface NavDataTypes {
  [x: string]: Array<NavLinks>;
}

interface NavTypes {
  areLinksVisible: boolean;
  isMenuOpen: boolean;
}

export type { NavDataTypes, NavTypes };
