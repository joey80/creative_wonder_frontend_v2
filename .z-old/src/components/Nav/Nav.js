import React from 'react';
import './Nav.scss';

const Nav = ({ areLinksVisible, isMenuOpen }) => (
  <div className={`nav nav--${isMenuOpen ? 'down' : 'hidden'}`}>
    <div className='nav__container'>
      <ul className={`nav__list__first ${areLinksVisible ? 'nav--down' : ''}`}>
        <li>Home</li>
        <li>About</li>
        <li>Examples</li>
        <li className='nav__arrow'>
          Blog
          <ul className='nav__list__second'>
            <li>Nintendo</li>
            <li>Sony</li>
            <li>Microsoft</li>
          </ul>
        </li>
        <li className='nav__arrow'>
          Gallery
          <ul className='nav__list__second'>
            <li>Go To Bed</li>
            <li>Brush Teeth</li>
            <li>Floss</li>
            <li>Jump Rope</li>
          </ul>
        </li>
        <li>Contact Us</li>
      </ul>
    </div>
  </div>
);

export default Nav;
