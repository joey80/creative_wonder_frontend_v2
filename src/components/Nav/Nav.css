/*

// ==========================================================================
// Nav Styles
// ==========================================================================

*/
.nav {
  position: absolute;
  top: 0;
  font-family: 'Lato', sans-serif;
  width: 100%;
  background-color: #15273f;
  color: rgba(255, 255, 255, 0);
  font-size: 1rem;
  transition: all .4s ease-out;
  overflow-x: hidden;
  z-index: 1; }
  @media (min-width: 768px) {
    .nav {
      margin-top: 80px;
      position: fixed;
      display: block;
      transform: translateY(0);
      overflow-x: visible;
      height: 60px;
      font-size: 0.95rem; } }
  .nav__container {
    max-width: 100%;
    margin: 0 auto;
    height: inherit; }
    @media (min-width: 768px) {
      .nav__container {
        max-width: 1170px; } }
  .nav a:link {
    text-decoration: none;
    color: inherit;
    background-color: transparent; }
  .nav a:visited {
    text-decoration: none;
    color: inherit;
    background-color: transparent; }
  .nav a:active {
    text-decoration: none;
    color: inherit;
    background-color: transparent; }
  .nav a:hover {
    text-decoration: none;
    color: inherit;
    background-color: transparent; }
  .nav__list__first {
    padding-top: 5px;
    padding-left: 20px;
    height: inherit;
    cursor: pointer;
    list-style-type: none;
    transform: translateY(-20px);
    transition: all .4s ease;
    color: transparent; }
    @media (min-width: 768px) {
      .nav__list__first {
        margin: 0;
        padding-top: 20px;
        display: inline-flex; } }
  .nav li:hover {
    color: #87bf7e; }
  .nav ul li {
    padding-bottom: 20px; }
    @media (min-width: 768px) {
      .nav ul li {
        padding-bottom: 0;
        padding-right: 48px; } }
  .nav ul ul {
    display: block;
    background-color: #1a304e;
    min-width: 800px;
    margin-left: -25px;
    margin-top: 10px; }
    @media (min-width: 768px) {
      .nav ul ul {
        position: absolute;
        display: none;
        margin: 14px auto;
        border-radius: 0 10px;
        width: auto;
        min-width: 3000px;
        margin-left: -25px;
        text-align: left; } }
  @media (min-width: 768px) {
    .nav ul li:hover > ul {
      color: rgba(255, 255, 255, 0.3);
      display: block; } }
  .nav ul ul li {
    list-style-type: none;
    padding: 15px 10px 15px 0; }
    @media (min-width: 768px) {
      .nav ul ul li {
        display: inline-flex;
        cursor: pointer;
        padding: 15px 20px 15px 0; } }
  .nav ul ul li:after {
    content: none;
    margin-left: 0; }
    @media (min-width: 768px) {
      .nav ul ul li:after {
        font-size: .8rem;
        font-weight: normal;
        font-style: normal;
        display: inline-block;
        margin-left: 20px;
        padding: 0;
        content: '|';
        text-decoration: inherit;
        color: rgba(251, 248, 248, 0.2); } }
  .nav ul ul li:last-child:after {
    content: none; }
  .nav__arrow:after {
    content: none; }
    @media (min-width: 768px) {
      .nav__arrow:after {
        font-family: FontAwesome;
        font-size: .4em;
        font-weight: normal;
        font-style: normal;
        display: inline-block;
        margin-left: 8px;
        content: '\f078';
        text-decoration: inherit; } }
  .nav--hidden {
    margin-top: 0;
    transition: all .4s ease;
    transform: translateY(-100%); }
  .nav--down {
    transition: all .4s ease-out;
    transform: translateY(0);
    color: #fff; }
