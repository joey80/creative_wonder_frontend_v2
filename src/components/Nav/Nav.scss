@import '../Global/Variables.scss';
@import '../Global/Mixins.scss';

/*

// ==========================================================================
// Nav Styles
// ==========================================================================

*/

.nav {
	display: none;
	top: 0;
	font-family: 'Lato', sans-serif;
	width: 100%;
	background-color: $primaryBlueColor;
	color: rgba(255, 255, 255, 0);
	font-size: 1rem;
	transition: all .4s ease-out;
	overflow-x: hidden;
	z-index: 1;

	@include tablet {
		margin-top: 80px;
		position: fixed;
		display: block;
		transform: translateY(0);
		overflow-x: visible;
		height: 60px;
		font-size: 0.95rem;
    }
    
    &__container {
        max-width: 100%;
        margin: 0 auto;

        @include tablet {
            max-width: 1170px;
        }
    }

	& a:link {
		text-decoration: none;
	    color: inherit;
	    background-color: transparent;
	}

	& a:visited {
		text-decoration: none;
	    color: inherit;
	    background-color: transparent;
	}

	& a:active {
		text-decoration: none;
	    color: inherit;
	    background-color: transparent;
	}

	& a:hover {
		text-decoration: none;
	    color: inherit;
	    background-color: transparent;
	}
	
	&__list__first {
		padding-top: 5px;
		padding-left: 20px;
 		height: inherit;
		cursor: pointer;
		list-style-type: none;
		transform: translateY(-20px);
        transition: all .4s ease;
        color: transparent;

		@include tablet {
			margin: 0;
			padding-top: 20px;
			display: inline-flex;
		}
	}

	& li:hover {
		color: $primaryGreenColor;
	}

	& ul li {
		padding-bottom: 20px;

		@include tablet {
			padding-bottom: 0;
			padding-right: 48px;
		}
	}

	& ul ul {
		display: block;
		background-color: $primaryLightBlueColor;
		min-width: 800px;
		margin-left: -25px;
		margin-top: 10px;

		@include tablet {
			position: absolute;
			display: none;
			margin: 9px auto;
			border-radius: 0 10px;
			width: auto;
			min-width: 3000px;
			margin-left: -25px;
			text-align: left;
		}
	}

	& ul li:hover > ul {

		@include tablet {
			color: rgba(255, 255, 255, 0.3);
			display: block;
		}
	}

	& ul ul li {
		list-style-type: none;
		padding: 15px 10px 15px 0;

		@include tablet {
			display: inline-flex;
			cursor: pointer;
			padding: 15px 20px 15px 0;
		}
	}

	& ul ul li:after {
		content: none;
		margin-left: 0;

		@include tablet {
			font-size: .8rem;
			font-weight: normal;
			font-style: normal;
			display: inline-block;
			margin-left: 20px;
			padding: 0;
			content: '|';
			text-decoration: inherit;
			color: rgba(251, 248, 248, 0.2);
		}
	}

	& ul ul li:last-child:after {
		content: none;
	}

	&__arrow:after {
		content: none;

		@include tablet {
			font-family: FontAwesome;
			font-size: .4em;
			font-weight: normal;
			font-style: normal;
			display: inline-block;
			margin-left: 8px;
			content: '\f078';
			text-decoration: inherit;
		}
	}

	&--hidden {
		transition: all .4s ease;

		@include tablet {
			margin-top: 0;
			transform: translateY(-100%);
		}
	}

	&--down {
		transition: all .4s ease-out;
		transform: translateY(0);
		color: #fff;
	}
}