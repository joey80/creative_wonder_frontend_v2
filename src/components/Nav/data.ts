import { NavDataTypes } from './types';

const data: NavDataTypes = {
  links: [
    { name: 'Home' },
    { name: 'About' },
    { name: 'Examples' },
    { name: 'Blog', children: [{ name: 'Nintendo' }, { name: 'Sony' }, { name: 'Microsoft' }] },
    {
      name: 'Gallery',
      children: [
        { name: 'Go To Bed' },
        { name: 'Brush Teeth' },
        { name: 'Floss' },
        { name: 'Jump Rope' },
      ],
    },
    { name: 'Contact Us' },
  ],
};

export { data };
