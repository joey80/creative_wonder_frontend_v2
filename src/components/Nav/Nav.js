import React, { Component } from 'react';
import './Nav.css';

class Nav extends Component {
  render() {
    return (
      <div className={`nav nav--${this.props.isMenuOpen ? "down" : "hidden"}`}>
        <div className="nav__container">
          <ul className={`nav__list__first ${this.props.areLinksVisible ? "nav--down" : ""}`}>
            <li>Home</li>
            <li>About</li>
            <li>Examples</li>
            <li class="nav__arrow">Blog
              <ul class="nav__list__second">
                <li>Nintendo</li>
                <li>Sony</li>
                <li>Microsoft</li>
              </ul>
            </li>
            <li class="nav__arrow">Gallery
              <ul class="nav__list__second">
                <li>Go To Bed</li>
                <li>Brush Teeth</li>
                <li>Floss</li>
                <li>Jump Rope</li>
              </ul>
            </li>
            <li>Contact Us</li>
          </ul>
        </div>
      </div>
    );
  }
}

export default Nav;
