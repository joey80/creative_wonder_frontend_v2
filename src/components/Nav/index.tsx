import React from 'react';
import { data } from './data';
import { NavTypes } from './types';
import './styles.scss';

const Nav = ({ areLinksVisible, isMenuOpen }: NavTypes): JSX.Element => (
  <div className={`nav nav--${isMenuOpen ? 'down' : 'hidden'}`}>
    <div className="nav__container">
      <ul className={`nav__list__first ${areLinksVisible ? 'nav--down' : ''}`}>
        {data.links.map(({ children, name }, index) => (
          <li className={children ? 'nav__arrow' : ''} key={index}>
            {name}
            {children ? (
              <ul className="nav__list__second">
                {children.map(({ name: innerName }, innerIndex) => (
                  <li key={innerIndex}>{innerName}</li>
                ))}
              </ul>
            ) : null}
          </li>
        ))}
      </ul>
    </div>
  </div>
);

export { Nav };
