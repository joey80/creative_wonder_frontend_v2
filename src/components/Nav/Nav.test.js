import React from 'react';
import Nav from './Nav';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const nav = renderer
    .create(<Nav />)
    .toJSON()
  expect(nav).toMatchSnapshot()
});
