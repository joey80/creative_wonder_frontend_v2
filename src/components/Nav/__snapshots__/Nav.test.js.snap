// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="nav"
>
  I'm the navbar component
</div>
`;
